From d11648b8f7d360a764926dfb7e57526f1db8977a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Fri, 3 Mar 2023 19:15:13 +0100
Subject: [PATCH] output: Check enable pending state when an enable commit is
 pending

This can happen in case of failure and we want to retry
it in this case.

Closes: https://gitlab.gnome.org/World/Phosh/phosh/-/issues/900
---
 src/output.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/output.c b/src/output.c
index 1763e21e..6c809187 100644
--- a/src/output.c
+++ b/src/output.c
@@ -1186,6 +1186,7 @@ phoc_output_handle_output_power_manager_set_mode (struct wl_listener *listener,
   struct wlr_output_power_v1_set_mode_event *event = data;
   PhocOutput *self;
   bool enable = true;
+  bool current;
 
   g_return_if_fail (event && event->output && event->output->data);
   self = event->output->data;
@@ -1203,7 +1204,11 @@ phoc_output_handle_output_power_manager_set_mode (struct wl_listener *listener,
     return;
   }
 
-  if (enable == self->wlr_output->enabled)
+  current = self->wlr_output->enabled;
+  if (self->wlr_output->pending.committed & WLR_OUTPUT_STATE_ENABLED)
+    current = self->wlr_output->pending.enabled;
+
+  if (enable == current)
     return;
 
   wlr_output_enable (self->wlr_output, enable);
-- 
GitLab

